next.config.js

```diff
@@ -59,6 +60,10 @@ module.exports = (phase, nextConfig) =>
         compress: process.env.COMPRESSION !== '0',
         env: {
             GA_TRACKING_ID: process.env.GA_TRACKING_ID,
-            SITE_URL: process.env.SITE_URL,
+            SITE_URL: Joi.attempt(process.env.SITE_URL,
+                Joi.string()
+                    .uri({ scheme: ['https', /http.*localhost/] })
```

Comment:
Why `/http.*localhost/`? Wasn't `http` sufficient?

Reply:
It was, but this is to ensure  HTTP is only allowed for `localhost`.

Reply:
I see! Well it's surprising that it works as `scheme` refers to the part before the `:`.

Reply:
Maybe we can use `try`?

```
Joi.alternatives().try(
    Joi.string().uri({ scheme: ['https'] }).pattern(/\/$/, { invert: true })
    Joi.pattern(/http:\/\/localhost:\d+/),
).message('process.env.SITE_URL must be a valid URL')),
```

Haven't tested.

Reply:
> I see! Well it's surprising that it works as `scheme` refers to the part before the `:`.

Oh, that's not in the documentation 🤔 hmm yes, it's working now because it's actually looking for the last instante of `:`, and because i was testing with `localhost:3000` it was working...

Reply:
We could change it, but i would also like some opinions about restricting it to `https`. I'm not sure if this is wise, i think it's ok to expect of development team to be able to use `https` but i'm not sure it's ok to expect other stakeholders to do so.

Reply:
I'm fine blocking it to `https` and only `http` for development. I'm just unsure if this is the right place to validate or do these kind of enforcement.

I would KISS and not enforce it as this level.

Reply:
Understood, will remove it for now.